adagrams/game.py

```diff
@@ -1,11 +1,107 @@
+import random
+
+# letters with point value and distribution
+LETTERS = {
+    "A" : [1, 9],
+    "B" : [3, 2],
+    "C" : [3, 2],
+    "D" : [2, 4],
+    "E" : [1, 12],
+    "F" : [4, 2],
+    "G" : [2, 3],
+    "H" : [4, 2],
+    "I" : [1, 9],
+    "J" : [8, 1],
+    "K" : [5, 1],
+    "L" : [1, 4],
+    "M" : [3, 2],
+    "N" : [1, 6],
+    "O" : [1, 8],
+    "P" : [3, 2],
+    "Q" : [10, 1],
+    "R" : [1, 6],
+    "S" : [1, 4],
+    "T" : [1, 6],
+    "U" : [1, 4],
+    "V" : [4, 2],
+    "W" : [4, 2],
+    "X" : [8, 1],
+    "Y" : [4, 2],
+    "Z" : [10, 1]
+}
+
+
 def draw_letters():
-    pass
+
+    # creates a pool of letters given distribution
+    letter_pool = []
+    for letter, stats in LETTERS.items():
+        letter_distribution = stats[1]
+        letter_pool += letter * letter_distribution
+    
+    # selects 10 letters from pool
+    user_letters = random.sample(letter_pool, k=10)
+
+    return user_letters
+    
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    # create a dictionary counting available letters
+    available_letters = {}
+    for letter in letter_bank:
+        available_letters[letter] = available_letters.get(letter, 0) + 1
+
+    for letter in word.upper():
+        # if letter is not available in bank, return False
+        if letter not in letter_bank or available_letters[letter] == 0:
+            return False
+        # if letter still available, subtract 1 from letter count
+        elif available_letters[letter] != 0:
+            available_letters[letter] -= 1
+
+    # return True if all used letters were available
+    return True
```

Comment:
Nice job using a dictionary to store the available letters, instead you could create a copy of the letter bank and iterate over it. Something like this:

```
def draw_letters():
    letter_pool_copy = copy.deepcopy(LETTER_POOL)
    letters = []

    while len(letters) < 10:
        letter = random.choice(list(letter_pool_copy.keys()))
        if letter_pool_copy[letter] > 0:
            letters.append(letter)
            letter_pool_copy[letter] -= 1

    return letters
```